src/improvements/tasks/MultisigTask.sol

```diff
@@ -187,9 +187,20 @@ abstract contract MultisigTask is Test, Script {
         public
         returns (VmSafe.AccountAccess[] memory)
     {
-        /// perform all actions in simulateRun, then send the transaction on chain
-        VmSafe.AccountAccess[] memory accountAccesses = simulateRun(taskConfigFilePath, signatures);
-        execute(signatures);
+        /// sets safe to the safe specified by the current template from addresses.json
+        _taskSetup(taskConfigFilePath);
+
+        /// gather mutative calls
+        build();
+
+        /// now execute task actions
+        VmSafe.AccountAccess[] memory accountAccesses = execute(signatures);
+
+        /// validate all state transitions
+        validate(accountAccesses);
+
+        /// print out results of execution
+        print();
```

Comment:
What is the motivation for this change? I think it's ok, just want to understand

Reply:
Got it. Are each of these bugs caught by one of the tests added in this PR? If not let's make sure to add that test before merging